templates/etc/systemd/system/caddy.service.erb

```diff
@@ -9,19 +9,20 @@ Wants=network-online.target systemd-networkd-wait-online.service
 User=<%= @caddy_user %>
 Group=<%= @caddy_group %>
 
-Environment=HOME=/etc/ssl/caddy
+Environment=HOME=<%= @caddy_home %>
+Environment=CADDYPATH=<%= @caddy_ssl_dir %>
 
-ExecStart=<%= @install_path %>/caddy -log <%= @caddy_log_dir %>/caddy.log -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp
+ExecStart=<%= @install_path %>/caddy -log <%= @caddy_log_dir %>/caddy.log -agree=true -conf=/etc/caddy/Caddyfile -root=/var/tmp -http-port=<%= @caddy_http_port %> -https-port=<%= @caddy_https_port %>
 ExecReload=/bin/kill -USR1 $MAINPID
 
 LimitNOFILE=1048576
-LimitNPROC=64
+LimitNPROC=512
 
 PrivateTmp=true
-PrivateDevices=true
+PrivateDevices=false
```

Comment:
why was this changed?

Reply:
I've had some troubles with the original configuration, this change slipped in while troubleshooting. Though, the recommended systemd file also disables PrivateDevices as this does not work on some devices.

ref: https://github.com/mholt/caddy/commit/fd3fafa50caf0dcbe695d28b48198a1e2bf810bd